### PR TITLE
Check server version in SDK

### PR DESCRIPTION
**Pull request by zhiltsov-max**
### Motivation and context

- Allowed unauthorized access to `api/about`, `api/docs`, `api/swagger`, `api/schema`
- Synchronized reported server API version in schema and in `api/about`
- Added tests
- Added a check in SDK and CLI for the server API version
  - Currently, versions from 2.0 to 2.3 are all considered supported
  - SDK allows to configure the desired check behaviour
- Fixed SDK .gitignore to ignore only files in the package root. It fixes file addition from `cvat-sdk/gen/`
- Improved host schema detection failure message (#4971)

### How has this been tested?

Unit tests

### Checklist

- [ ] I submit my changes into the `develop` branch
- [ ] I have added a description of my changes into [CHANGELOG](https://github.com/cvat-ai/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/cvat-ai/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))

### License

- [ ] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

File: cvat-sdk/cvat_sdk/core/client.py [hunk 1]
```diff
@@ -162,6 +182,44 @@ def wait_for_completion(
...
+                "Some SDK functions may not work properly with this server. "
+                "You can continue using this SDK, or you can "
+                "try to update with 'pip install cvat-sdk'."
+            ) % (server_version, sdk_version)
+            self.logger.warning(msg)
+            if fail_if_unsupported:
+                raise IncompatibleVersionException(msg)
+
+    def get_server_version(self) -> pv.Version:
+        # TODO: allow to use this endpoint unauthorized
```
Comment (nmanovic) [inline cvat-sdk/cvat_sdk/core/client.py:219]: Probably need to remove the TODO

**Review by nmanovic — APPROVED** [review]
LGTM

**Comment by nmanovic** [comment]
@zhiltsov-max , could you please look at tests?

**Comment by kirill-sizov** [comment]
Tests are red because this branch doesn't contain last changes from develop branch, please try to merge `develop` branch into the `zm/check-server-version-in-sdk`

**Pull request merged**